Introduce StepContainerV2

zaguiini opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/98648.

## Proposed Changes

https://github.com/user-attachments/assets/447d1519-e01d-4c6a-a5f3-78cf06acd076




## Why

Standardization and bug solving. Here's an example of a bug that got introduced after some code removal with the goal of standardization: https://github.com/Automattic/wp-calypso/pull/99944. It's too difficult to track how changes will affect steps because of the gazillion customizations we have. Starting from scratch with no class clashes and customizations will ensure we'll get to the desired state.

### Treat StepContainer as a low-level component

Instead, the onboarding package should expose wireframes like `<TwoColumnLayout />` that uses the `<StepContainer />` behind the scenes to render everything.

### Composable wireframes

`StepContainerV2` exposes all primitives as components so you can create wireframes at will.

## Testing Instructions

Open Storybook 👍🏻 
`cd packages/onboarding && yarn storybook`

zaguiini committed:
Create a composable step container

zaguiini committed:
Add HorizontalStepLayout component

zaguiini committed:
Add CenteredStepLayout component

zaguiini committed:
Record Tracks events on back and skip click

zaguiini committed:
Fix WPCOM wordmark width

zaguiini committed:
Match Next button font size with designs

zaguiini committed:
Fix top bar height

zaguiini committed:
Accept ReactNode in heading.text and heading.subText

zaguiini committed:
Fix Storybook

zaguiini committed:
Refine stories

zaguiini committed:
Dedupe Yarn packages

zaguiini committed:
Make StepContainerV2 more composable

zaguiini committed:
Add shouldUseStepContainerV2 helper

zaguiini committed:
Introduce SixColumnsCenteredStepLayout

zaguiini committed:
Export StepContainerV2 items as a namespace

zaguiini committed:
Allow removing content padding

zaguiini committed:
Fix padding settings

zaguiini committed:
Fix WordPressWordmark disposition

p-jackson committed:
Memoize context

p-jackson committed:
Fix horizontal scrollbar appearing on Windows

p-jackson committed:
Translate stepper button labels

p-jackson reviewed (approved):
Storybook is looking good. Excited to see how this works out once we start adding it to steps!

autumnfjeld commented:
cc: @Automattic/vertex  on this work to enforce style consistency.

p-jackson committed:
Further adjust checkout wordmark size

I made it an even closer size match to what exists in production. Medium
term we probably want the checkout workmark to match the size used in
the onboarding header. But in the short term this PR isn't supposed to
change anything about the checkout.

p-jackson commented:
I made a slight tweak to ensure the size of the wordmark in the checkout is even closer to the way it was before (although the alignment with the close button is slightly better now). This PR should just be adding boilerplate, not making any visible changes.

In the future we probably want the checkout wordmark to be the same size as the one used in the onboarding header. But we haven't shipped those, so not yet.

| Production | This PR |
| --- | --- |
| ![before](https://github.com/user-attachments/assets/ded06c82-bfd2-4f43-a113-b2949c47fc41) | ![after](https://github.com/user-attachments/assets/c3e8d160-08b9-496f-bed8-87722197b446) |
